util/util.go

```diff
@@ -113,20 +113,95 @@ func GetBlockRangeInPages(ctx context.Context, from, to, pageSize uint64, c *eth
 }
 
 func GetReceipts(ctx context.Context, rawBlocks []*json.RawMessage, c *ethrpc.Client, batchSize uint64) ([]*json.RawMessage, error) {
+	// check if get receipts by block is supported
+	err := c.CallContext(ctx, nil, "eth_getBlockReceipts", "0x1")
```

Comment:
I think the value of the test to see if `eth_getBlockReceipts` is supported should be cached. No reason to make this call multiple times.